ApiBundle/Transformer/NodeGroupRoleTransformer.php

```diff
@@ -2,105 +2,55 @@
 
 namespace OpenOrchestra\ApiBundle\Transformer;
 
-use OpenOrchestra\ApiBundle\Exceptions\HttpException\RoleNotFoundHttpException;
-use OpenOrchestra\ApiBundle\Exceptions\TransformerParameterTypeException;
 use OpenOrchestra\Backoffice\Collector\RoleCollectorInterface;
 use OpenOrchestra\Backoffice\Model\GroupInterface;
-use OpenOrchestra\Backoffice\Model\NodeGroupRoleInterface;
 use OpenOrchestra\BaseApi\Facade\FacadeInterface;
-use OpenOrchestra\BaseApi\Transformer\AbstractTransformer;
 use OpenOrchestra\BaseBundle\Context\CurrentSiteIdInterface;
+use OpenOrchestra\ModelInterface\Model\NodeInterface;
 use OpenOrchestra\ModelInterface\Repository\NodeRepositoryInterface;
 
 /**
  * Class NodeGroupRoleTransformer
  */
-class NodeGroupRoleTransformer extends AbstractTransformer implements TransformerWithGroupInterface
+class NodeGroupRoleTransformer extends ModelGroupRoleTransformer
 {
-    protected $nodeRoleGroupClass;
+    protected $modelGroupRoleClass;
     protected $collector;
     protected $nodeRepository;
     protected $currentSiteManager;
 
     /**
      * @param string                  $facadeClass
-     * @param string                  $nodeRoleGroupClass
+     * @param string                  $modelGroupRoleClass
      * @param RoleCollectorInterface  $collector
      * @param NodeRepositoryInterface $nodeRepository
      * @param CurrentSiteIdInterface  $currentSiteManager
      */
     public function __construct(
         $facadeClass,
-        $nodeRoleGroupClass,
+        $modelGroupRoleClass,
         RoleCollectorInterface $collector,
         NodeRepositoryInterface $nodeRepository,
         CurrentSiteIdInterface $currentSiteManager
     ) {
-        parent::__construct($facadeClass);
-        $this->nodeRoleGroupClass = $nodeRoleGroupClass;
-        $this->collector = $collector;
+        parent::__construct($facadeClass, $modelGroupRoleClass, $collector);
         $this->nodeRepository = $nodeRepository;
         $this->currentSiteManager = $currentSiteManager;
     }
 
     /**
-     * @param NodeGroupRoleInterface $nodeGroupRole
+     * @param GroupInterface  $group
+     * @param FacadeInterface $facade
      *
-     * @return FacadeInterface
-     *
-     * @throws TransformerParameterTypeException
+     * @return bool
      */
-    public function transform($nodeGroupRole)
+    protected function isParentGranted(GroupInterface $group, FacadeInterface $facade)
```

Comment:
I'd prefer something like hasAccessOnParent or isAccessGrantedOnParent or something like this because it's not the parent who is granted but the user :)